pins/tests/test_boards.py

```diff
@@ -304,6 +310,47 @@ def test_board_pin_read_insecure_succeed_board_flag(board):
         board.pin_read("test_pin")
 
 
+# pin_write with unversioned boards ===========================================
+
+
+@pytest.mark.parametrize("versioned", [None, True, False])
+def test_board_unversioned_pin_write_once_unversioned(versioned, board_unversioned):
+    board_unversioned.pin_write({"a": 1}, "test_pin", type="json", versioned=versioned)
+
+    assert len(board_unversioned.pin_versions("test_pin")) == 1
+
+
+@pytest.mark.parametrize("versioned", [None, False])
+def test_board_unversioned_pin_write_unversioned(versioned, board_unversioned):
+    board_unversioned.pin_write({"a": 1}, "test_pin", type="json", versioned=versioned)
+    board_unversioned.pin_write({"a": 2}, "test_pin", type="json", versioned=versioned)
+
+    assert len(board_unversioned.pin_versions("test_pin")) == 1
+
+
+def test_board_unversioned_pin_write_versioned(board_unversioned):
+    board_unversioned.pin_write({"a": 1}, "test_pin", type="json", versioned=False)
+    board_unversioned.pin_write({"a": 2}, "test_pin", type="json", versioned=True)
+
+    assert len(board_unversioned.pin_versions("test_pin")) == 2
+
+
+def test_board_versioned_pin_write_unversioned(board):
+    # should fall back to the versioned setting of the board
+    board.pin_write({"a": 1}, "test_pin", type="json", versioned=False)
```

Comment:
If you pin_write(..., versioned=False) a second time, will it still just be 1 version? It feels like that might be a critical piece of this test (since just 1 write should always be 1 version, regardless of versioned?).

Reply:
ah, yeah I don't think this test is meaningful, so I pulled it. The intention was that 1-write pins don't error, etc, but it's true that is tested elsewhere!